tests/mocha/contextmenu_items_test.js

```diff
@@ -0,0 +1,430 @@
+        chai.assert.isTrue(block2.isCollapsed(),
+            'Previously expanded block should now be collapsed');
+      });
+
+      test('Has correct label', function() {
+        chai.assert.equal(collapseOption.displayText(), 'Collapse Blocks');
+      });
+    });
+
+    suite('Expand', function() {
+      var expandOption;
+      setup(function() {
+        expandOption = registry.getItem('expandWorkspace');
+      });
+
+      test('Enabled when collapsed blocks', function() {
+        workspace.newBlock('text');
+        var block2 = workspace.newBlock('text');
+        block2.setCollapsed(true);
+
+        chai.assert.equal(expandOption.preconditionFn(scope), 'enabled',
+            'Should be enabled when any blocks are collapsed');
+      });
+
+      test('Disabled when no collapsed blocks', function() {
+        workspace.newBlock('text');
+        chai.assert.equal(expandOption.preconditionFn(scope), 'disabled',
+            'Should be disabled when no blocks are collapsed');
+      });
+
+      test('Hidden when no collapse option', function() {
+        var workspaceWithOptions = new Blockly.Workspace(new Blockly.Options({collapse: false}));
+        scope.workspace = workspaceWithOptions;
+
+        chai.assert.equal(expandOption.preconditionFn(scope), 'hidden',
+            'Should be hidden if collapse is disabled in options');
+
+        workspaceWithOptions.dispose();
+      });
+
+      test('Expands all blocks', function() {
+        // All blocks should be expanded, even if some already were.
+        var block1 = workspace.newBlock('text');
+        var block2 = workspace.newBlock('text');
+        // Need to render block to properly collapse it.
+        block2.initSvg();
+        block2.render();
+        block2.setCollapsed(true);
+
+        expandOption.callback(scope);
+        this.clock.runAll();
+
+        chai.assert.isFalse(block1.isCollapsed(),
+            'Previously expanded block should still be expanded');
+        chai.assert.isFalse(block2.isCollapsed(),
+            'Previously collapsed block should now be expanded');
+      });
+
+      test('Has correct label', function() {
+        chai.assert.equal(expandOption.displayText(), 'Expand Blocks');
+      });
+    });
+
+    suite('Delete', function() {
+      var deleteOption;
+      setup(function() {
+        deleteOption = registry.getItem('workspaceDelete');
+      });
+
+      test('Enabled when blocks to delete', function() {
+        workspace.newBlock('text');
+        chai.assert.equal(deleteOption.preconditionFn(scope), 'enabled');
+      });
+
+      test('Disabled when no blocks to delete', function() {
+        chai.assert.equal(deleteOption.preconditionFn(scope), 'disabled');
+      });
+
+      test('Deletes all blocks after confirming', function() {
+        // Mocks the confirmation dialog and calls the callback with 'true' simulating ok.
+        sinon.stub(Blockly, 'confirm').callsArgWith(1, true);
```

Comment:
nice!
I would also add an assert that the confirm dialog was called.
```
var confirmStub = sinon.stub(Blockly, 'confirm');
confirmStub.callsArgWith(1, true);
...
sinon.assert.calledOnce(confirmStub);
```

Same comment for other instances where you are stubbing confirm.

Reply:
Done